pkg/controller/master/setting/additional_guest_memory_overhead.go

```diff
@@ -0,0 +1,71 @@
+package setting
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+// upstream CRD field path Spec.Configuration.AdditionalGuestMemoryOverheadRatio
+const overheadConfigName = "AdditionalGuestMemoryOverheadRatio"
+
+func (h *Handler) syncAdditionalGuestMemoryOverheadRatio(setting *harvesterv1.Setting) error {
+	var value string
+	if setting.Value != "" {
+		value = setting.Value
+	} else {
+		value = setting.Default
+	}
+	logrus.Debugf("update %v as %v to kubevirt", overheadConfigName, value)
+	agmorc, err := settings.NewAdditionalGuestMemoryOverheadRatioConfig(value)
+	if err != nil {
+		return err
+	}
+
+	// sync with  kubevirt configuration
+	kubevirt, err := h.kubeVirtConfigCache.Get(util.HarvesterSystemNamespaceName, util.KubeVirtObjectName)
+	if err != nil {
+		return fmt.Errorf("fail to get kubevirt object %v/%v", util.HarvesterSystemNamespaceName, util.KubeVirtObjectName)
+	}
+	// kubevirt already has AdditionalGuestMemoryOverheadRatio
+	if kubevirt.Spec.Configuration.AdditionalGuestMemoryOverheadRatio != nil {
+		// clear current
+		if agmorc.IsEmpty() {
```

Comment:
the default additional overhead will never be empty because when setting does have not a custom value, we switch to default of 1.5

a user who may have manually updated their configuration, will end up having this changed to 1.5 which may not be desirable.

may be we should leave the default value to an empty string?

Reply:
Good questions & suggestions :)

(1) For newly installed cluster or upgraded cluster, if user does not touch this setting; the default "1.5" is designed to replace the legacy "100Mi" reserved memory.

(2) When the `Default` is `""`, this feature means to be disabled, and user needs to set it manually to get this feature.

Roughly, (1) will benefit more users.


(3) If a user doesn't want this setting but fully fallback to legacy `Reserved Memory` mode, he can set this setting as "0"; this is described in the document PR.

![image](https://github.com/user-attachments/assets/f5178f5a-3366-4cbe-ae9f-c7197b439490)

Reply:
in the kubevirt spec AdditionalGuestMemoryOverheadRatio defaults to 1.

If say a user had a kubevirt config modified with this set, the code block will find `kubevirt.Spec.Configuration.AdditionalGuestMemoryOverheadRatio` to be not nil.

In subsequent step `agmorc.IsEmpty()` is never going to be empty since we are using `Default` of 1.5 when a `Value` is not provided in the setting, which will trigger the update the wipe the user defined ration and set it to 1.5 which may not be desired.

![image](https://github.com/user-attachments/assets/d46ffeee-2923-4574-bdac-f58e35b42d47)

Reply:
When Harvester expose this option to setting, we generally assume the setting is the source of truth, changes to underlayer will be overwritten.

I will add this to document, if users has used to set this on kubevirt object directly, then user is suggested to set on Harvester settings later.

On the upgrade path, will also add another PR to convert the potential user set value to Harvester setting.